### PR TITLE
OSLC Automation PSD01 draft

### PR DESCRIPTION
**Pull request by jamsden**
Getting the Automation spec so that it can be published as a PSD in order to be referenced.

### CODE REVIEW

**Comment by jamsden** [comment]
@ndjc I'm getting a ReSpec error that the conformance section is missing (and indeed it is) in automation-spec.html. But the section with id="conformance" is there. I can't find anything wrong. Can you take a look?

File: specs/auto/automation-shapes.html [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+          },
+        },
+      };
+    </script>
+  </head>
+
+  <body>
+    <section id="abstract">
+      <p>
+        OSLC Automation Version 2.1 Part 2: Vocabulary defines the OSLC Automation RDF vocabulary terms and resources.
```
Comment (berezovskyi) [inline specs/auto/automation-shapes.html:173]: Part 3

File: specs/auto/automation-shapes.html [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+          title="Shape for oslc_auto:ParameterInstance"
+          data-include="./automation-shapes.ttl#ParameterInstanceShape"
+          data-oninclude="shapeToSpec"
+          data-include-sync="true"
+          data-include-replace="true"
+          data-include-format="html"
+        ></div>
+      </section>
+
+      <section id="Resource_Dialog">
```
Comment (berezovskyi) [inline specs/auto/automation-shapes.html:337]: I think the id has to be DialogShape for proper HTML conneg link destinations

**Review by berezovskyi — CHANGES_REQUESTED** [review]

**Comment by ndjc** [comment]
>  I'm getting a ReSpec error that the conformance section is missing (and indeed it is) in automation-spec.html. But the section with id="conformance" is there. I can't find anything wrong. Can you take a look?

@jamsden the problem is that you have specified `noConformanceTable = true` in the configuration section, but you have not specified your own conformance section **text**. You must either use the auto-generated table, or provide your own text. For example, in your Core file `core-shapes.ttl` you have `noConformanceTable = true` but with the following explicit conformance section with content:

>     <section id="conformance">
>       <h2>Conformance</h2>
>       <p>OSLC servers MUST follow the constraints defined here where required, and with the meanings defined here.</p>
>       <p>OSLC servers MAY provide additional constraints for specific purposes.</p>
>     </section>

**Review by ndjc — APPROVED** [review]
Changes to conformance table look good.
I see at least two missing vocabulary terms:

1.  The enum value `oslc_auto:passed`. This is a verdict, like `error`, `fail`, `unavailable`, or `warning`. All those are in the vocabulary, but `passed` is missing.
2. The property predicate `oslc_auto:progress`. This is one of the new terms needed by OSLC Configuration Management. I'll send a draft property definition separately.

I say at least two, because there are several ShapeChecker errors. While most of these are missing or incorrect parts of otherwise good property definitions, there are some that point to inconsistencies in the way parameters are defined. For example, the shapes use `:inputParameter` and `:outputParameter`, but the vocabulary defined `:additionalParameter`.

While all these issues need to be fixed, I'm approving the PR so that we can have a first version of the draft in the master branch.

**Comment by ndjc** [comment]
Here's an OSLC property definition for the oslc_auto:progress property:

>         [
>             #progress
>             a                        oslc:Property ;
>             oslc:name                "progress" ;
>             oslc:propertyDefinition  oslc_auto:progress ;
>             oslc:occurs              oslc:Zero-or-one ;
>             oslc:readOnly            true ;
>             oslc:valueType           xsd:integer ;
>             dcterms:description      "The percentage (0-100) of completion, if known."^^rdf:XMLLiteral
>         ] ,

Here's a possible vocabulary entry:

>        <http://open-services.net/ns/auto#progress>
>               a                 rdf:Property ;
>               rdfs:comment      "A percentage (0-100) of completion." ;
>               rdfs:isDefinedBy  <http://open-services.net/ns/auto#> ;
>               rdfs:label        "progress"  .

**Review by berezovskyi — APPROVED** [review]

**Comment by berezovskyi** [comment]
image

@jamsden press "update the base branch" to solve the lint build issues before merge, I fixed them in master

**Pull request merged**